tests/test_searchspace.py

```diff
@@ -86,6 +88,31 @@ def test_discrete_searchspace_creation_from_dataframe():
     assert df.equals(searchspace.discrete.exp_rep)
 
 
+def test_invalid_discrete_searchspace_creation_from_simplex():
+    """A purely discrete search space is created from a simplex."""
```

Comment:
test name and docstring are way to generic here. They should reflect what is actually being tested, i.e. you can almost use the comment from inside the test here